services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistorySpec.groovy

```diff
@@ -0,0 +1,139 @@
+        flowH3[1].taskId
+        flowH3[1].histories.action[0] == "Updated the flow"
+    }
+
+    def "History records are created for the create/update actions using default timeline"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        then: "History record is created"
+        def flowH1 = northbound.getFlowHistory(flow.id)
+
+        flowH1.size() == 1
+        flowH1[0].flowId == flow.id
+        flowH1[0].action == "Flow creating"
+        flowH1[0].taskId
+        flowH1[0].histories.action[0] == "Created the flow"
+
+        when: "Update the created flow"
+        flowHelper.updateFlow(flow.id, flow.tap { it.description = it.description + "updated" })
+
+        then: "History record is created"
+        def flowH2 = northbound.getFlowHistory(flow.id)
+        flowH2.size() == 2
+        flowH2[1].flowId == flow.id
+        flowH2[1].action == "Flow updating"
+        flowH2[1].taskId
+        flowH2[1].histories.action[0] == "Updated the flow"
+
+        when: "Delete the updated flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "History is still available for the deleted flow"
+        def flowH3 = northbound.getFlowHistory(flow.id)
+        flowH3.size() == 2
+        flowH1[0].flowId == flow.id
+        flowH3[0].action == "Flow creating"
+        flowH3[0].taskId
+        flowH3[0].histories.action[0] == "Created the flow"
+
+        flowH1[0].flowId == flow.id
+        flowH3[1].action == "Flow updating"
+        flowH3[1].taskId
+        flowH3[1].histories.action[0] == "Updated the flow"
+    }
+
+    def "History should not be returned in case timeline is incorrect (beforetime > afterTime)"() {
+        given: "Current timestamp"
+        Long timestampBefore = (System.currentTimeMillis() / 1000L) - 1000
+
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        then: "History record is created"
+        Long timestampAfterCreate = System.currentTimeMillis() / 1000L
+        def flowH1 = northbound.getFlowHistory(flow.id, timestampBefore, timestampAfterCreate)
+
+        flowH1.size() == 1
+        flowH1[0].flowId == flow.id
+        flowH1[0].action == "Flow creating"
+        flowH1[0].taskId
+        flowH1[0].histories.action[0] == "Created the flow"
+
+        when: "Try to get history for incorrect timeline"
+        def flowH2 = northbound.getFlowHistory(flow.id, timestampAfterCreate, timestampBefore)
+
+        then: "History record is NOT returned"
+        flowH2.isEmpty()
+
+        and: "cleanup: Restore default state(remove created flow)"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "History should not be returned in case flowId is not being created yet"() {
```

Comment:
in case of a non-existent flow

Reply:
I thought about it.
in a nutshell history is still available for non-existent flow(I mean flow can be deleted)

this test checks history shouldn't be returned in case for flow which was never being created

Reply:
I propose just to rename 
"History should not be returned in case flowId is not being created yet"
->
"History should not be returned in case flow was never created"